adjustments to objdump call

GitMensch opened the pull request:
After recognizing that the process path of objdump is resolved first, I've adjusted the PR to use that (so the additional `stat` calls seen in #527 are not needed).

* invoke directly from the processPath already resolved before
* improve error message for objdump invocation issues helping to find problems related to the call

lievenhey commented:
@GitMensch lgtm, but can you ran `clang-format` so that the ci no longer complains?

GitMensch commented:
applied `clang-format -i` and pushed

milianw commented on src/models/disassemblyoutput.cpp:
```diff
@@ -308,22 +308,26 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
 
     if (!asmProcess.waitForStarted()) {
-        disassemblyOutput.errorMessage += QApplication::translate("DisassemblyOutput", "Process was not started.");
+        disassemblyOutput.errorMessage += QApplication::translate("DisassemblyOutput", "Process was not started.")
```
string puzzle, this is impossible to translate properly. Make this one large string and then use arg to fill in the placeholders and also include the `QProcess::errorString`

```
..., "<qt>Process failed to start: <tt>%1 %2</tt> returned <tt>%2</tt>").arg(processPath, arguments.join(...), asmProcess.errorString())
```

milianw commented on src/models/disassemblyoutput.cpp:
```diff
@@ -308,22 +308,26 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     if (!asmProcess.waitForFinished()) {
         disassemblyOutput.errorMessage +=
-            QApplication::translate("DisassemblyOutput", "Process was not finished. Stopped by timeout");
+            QApplication::translate("DisassemblyOutput", "Process was not finished. Stopped by timeout")
```
dito

GitMensch replied:
will fix the string puzzles, I likely was concerned about not breaking the existing translations

GitMensch commented:
* string puzzle fixed, other invocations of error messages changed accordingly
* additional commit fixing a related issue (missing check -> error message)

milianw commented on src/models/disassemblyoutput.cpp:
```diff
@@ -272,7 +272,9 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     if (processPath.isEmpty()) {
         disassemblyOutput.errorMessage =
-            QApplication::tr("Cannot find objdump process %1, please install the missing binutils package for arch %2")
+            QApplication::translate("DisassemblyOutput",
```
these hunks here are unrelated to what the commit message talks about - split that up into a separate commit

milianw commented on src/models/disassemblyoutput.cpp:
```diff
@@ -303,27 +305,32 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     arguments.append(binary);
 
-    asmProcess.start(objdump, arguments);
+    asmProcess.start(processPath, arguments);
```
ideally this here is also its own patch - whenever your commit message is something like "do A and B" then it's an indication that you should have split it up into two patches instead

milianw commented on src/models/disassemblyoutput.cpp:
```diff
@@ -265,7 +265,15 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     disassemblyOutput.symbol = symbol;
     if (symbol.symbol.isEmpty()) {
-        disassemblyOutput.errorMessage = QApplication::tr("Empty symbol ?? is selected");
+        disassemblyOutput.errorMessage =
```
this doesn't really help - if you really want to wrap the `??` then just do that directly, no need for `.arg()`

GitMensch replied:
That's what I did, just to get a compile warning about trigraphs., which was the reason to split that. Another option is to use inline hex values, bit that is likely not useful for the translation.

GitMensch commented:
rebased and split from 2 to 3 separate commits (I do like the result more than the original, too); tested manually in the GUI using the files under "tests" (all but the timeout, checked this with the appImage and also works as expected).

Adjusted one test as needed and added some of the missing tests for the adjusted messages.

milianw commented:
thanks!
